src/NServiceBus.Core.Tests/Pipeline/MutateTransportMessage/MutateIncomingTransportMessageBehaviorTests.cs

```diff
@@ -12,20 +12,92 @@ public class MutateIncomingTransportMessageBehaviorTests
         public void Should_throw_friendly_exception_when_IMutateIncomingTransportMessages_MutateIncoming_returns_null()
         {
             var behavior = new MutateIncomingTransportMessageBehavior();
-            
+
             var context = new TestableIncomingPhysicalMessageContext();
 
             context.Builder.Register<IMutateIncomingTransportMessages>(() => new MutateIncomingTransportMessagesReturnsNull());
 
             Assert.That(async () => await behavior.Invoke(context, () => TaskEx.CompletedTask), Throws.Exception.With.Message.EqualTo("Return a Task or mark the method as async."));
         }
 
+        [Test]
+        public async Task When_no_mutator_updates_the_body_should_not_update_the_body()
+        {
+            var behavior = new MutateIncomingTransportMessageBehavior();
+
+            var context = new InterceptUpdateMessageIncomingPhysicalMessageContext();
+
+            context.Builder.Register<IMutateIncomingTransportMessages>(() => new MutatorWhichDoesNotMutateTheBody());
+
+            await behavior.Invoke(context, () => TaskEx.CompletedTask);
+
+            Assert.False(context.UpdateMessageBodyCalled);
+        }
+
+        [Test]
+        public async Task When_no_mutator_available_should_not_update_the_body()
+        {
+            var behavior = new MutateIncomingTransportMessageBehavior();
+
+            var context = new InterceptUpdateMessageIncomingPhysicalMessageContext();
+
+            context.Builder.Register(() => new IMutateIncomingTransportMessages[]{ });
+
+            await behavior.Invoke(context, () => TaskEx.CompletedTask);
+
+            Assert.False(context.UpdateMessageBodyCalled);
+        }
+
+        [Test]
+        public async Task When_mutator_modifies_the_body_should_update_the_body()
+        {
+            var behavior = new MutateIncomingTransportMessageBehavior();
+
+            var context = new InterceptUpdateMessageIncomingPhysicalMessageContext();
+
+            context.Builder.Register<IMutateIncomingTransportMessages>(() => new MutatorWhichMutatesTheBody());
+
+            await behavior.Invoke(context, () => TaskEx.CompletedTask);
+
+            Assert.True(context.UpdateMessageBodyCalled);
+        }
```

Comment:
very minor nitpicking, but it constantly hit my eye while going through the tests: there is an empty line around every statement which doesn't seem to really add any value towards readability. Just my opinion, non-blocking comment.